formats/gltf/model.go

```diff
@@ -51,3 +51,86 @@ type PolyformTexture struct {
 	URI     string
 	Sampler *Sampler
 }
+
+func (pm PolyformMaterial) equal(other PolyformMaterial) bool {
+	if pm.Name != other.Name {
```

Comment:
An argument can be made that `Name` doesn't matter. They are arbitrary and optional and thus - don't need to be checked. I opted to check this as well, but practically it's of no use for the rendering process, so we could omit it for brevity. Lmk if you think it should be omitted.

Reply:
Fair point. I can't foresee why someone would care about a name, but I also don't want to potentially mess up an export for someone who builds logic around the material definition/name.

I think if we do remove this name check, we should introduce a new argument in the `NewWriter()` function, which would be a `WriterConfig` struct. I think for the time being, it'll have one field that'd be like, `OptimizeMaterials `. But it will allow us to add more options over time without breaking the API. If `OptimizeMaterials` is set to true, we do everything you've created in this PR. If it's set to false, it doesn't do anything

```golang
type WriterConfig struct {
    OptimizeMaterials bool
}
```

I might do this anyway. It would be a valid spot to add a new field like `PanicOnInvalidDefinition`, which would have addressed yesterdays PR.

Lemme know your thoughts!

Reply:
I would leave it as it is implemented now. 

I assume in 99.9% of cases `OptimizeMaterials` is the desired behavior anyway. There is a microscopic performance hit on having to run the deduplication, but it's negligible, while the benefit of reduced model size is material. I struggle to imagine a use case where duplicated materials would be preferable. So putting it behind a flag that needs to be manually set, just to have it set in 99.9% of the times would just create additional lines in the caller implementation for no good reason.

If for whatever reason the user does want to have duplicates - the current shape of implementation gives them that ability. Assign unique names to materials, and they will be duplicated.